core/iwasm/common/wasm_c_api.c

```diff
@@ -3956,7 +3956,6 @@ interp_link(const wasm_instance_t *inst, const WASMModule *module_interp,
             case IMPORT_KIND_MEMORY:
             case IMPORT_KIND_TABLE:
                 ASSERT_NOT_IMPLEMENTED();
```

Comment:
This leads to compilation warning in debug mode: `warning: this statement may fall through`
How about changing to:
```C
            case IMPORT_KIND_MEMORY:
            case IMPORT_KIND_TABLE:
                ASSERT_NOT_IMPLEMENTED();
                goto handle_default;
            handle_default:
            default:
```

Reply:
it's unfortunate that clang doesn't recognize the traditional `/* FALLTHROUGH */`.
how do you think about introducing something like `#defile BH_FALLTHROUGH __attribute__((fallthrough))`?

Reply:
Had better not, it might be not friendly to some compilers.
We have the similar case in:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/compilation/aot_emit_numberic.c#L455-L458
The modification such as that should be OK.

Or how about handling default together with `case IMPORT_KIND_MEMORY/IMPORT_KIND_TABLE`:
```C
            case IMPORT_KIND_MEMORY:
            case IMPORT_KIND_TABLE:
            default:
                  ASSERT_NOT_IMPLEMENTED();
                  LOG_WARNING("%s meets unsupported kind: %d", __FUNCTION__,
                              import_rt->kind);
                  goto failed;
```

Reply:
> Had better not, it might be not friendly to some compilers. We have the similar case in: https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/compilation/aot_emit_numberic.c#L455-L458 The modification such as that should be OK.

ok

> 
> Or how about handling default together with `case IMPORT_KIND_MEMORY/IMPORT_KIND_TABLE`:
> 
> ```c
>             case IMPORT_KIND_MEMORY:
>             case IMPORT_KIND_TABLE:
>             default:
>                   ASSERT_NOT_IMPLEMENTED();
>                   LOG_WARNING("%s meets unsupported kind: %d", __FUNCTION__,
>                               import_rt->kind);
>                   goto failed;
> ```

i think it makes the most sense this particular case. done.